src/test/create-user.test.ts

```diff
@@ -0,0 +1,112 @@
+import axios from "axios";
+import { expect } from "chai";
+import { dataSource } from "../data-source";
+import { User } from "../entity/User";
+import * as bcrypt from "bcrypt";
+
+describe("Mutation createUser", () => {
+  afterEach(async function () {
+    await dataSource.query('TRUNCATE TABLE "user"');
+  });
+
+  const mutation = `mutation CreateUser($data: UserInput!) {
+        createUser(data: $data) {
+          id
+          email
+          name
+          birthDate
+        }
+      }
+      `;
+  const urlDB = "http://localhost:4000/graphql";
+  let input = {
+    name: "Samuel Satana",
+    email: "SamuelTeste1@gmail.com",
+    birthDate: "21/02/2002",
+    password: "1234qwer",
+  };
+
+  it("should return createUser", async () => {
+    const response = await axios.post(urlDB, {
+      variables: {
+        data: input,
+      },
+      query: mutation,
+    });
+    const userDb = await dataSource.findOneBy(User, {
+      email: input.email,
+    });
+    const passwordCompare = await bcrypt.compare(
+      input.password,
+      userDb.password
+    );
+
+    const expectedResponse = {
+      data: {
+        createUser: {
+          id: userDb.id,
+          email: input.email,
+          name: input.name,
+          birthDate: input.birthDate,
+        },
+      },
+    };
+    expect(response.status).to.equal(200);
+    expect(response.data).to.be.deep.eq(expectedResponse);
+    expect(userDb.name).to.be.deep.eq(expectedResponse.data.createUser.name);
```

Comment:
outro ponto é que vc pode dar um `expect({objeto que vc quer}).to.be.deep.eq(expectedResponse)`, assim não precisa comparar termo a termo

Reply:
Aqui eu estava comparando se o usuário criado no banco é o mesmo do input, se eu pegar o objeto completo do usuário no banco ele trás junto o password mas para o password eu já tenho outro test usando o compare do bycript, e o password eu não tenho no response.data, ai não sei se teria como pegar o userDb sem o campo do password pra comparar?!